addressfinder.js

```diff
@@ -194,27 +209,64 @@ var fieldsForAddressType = {
   /*
    * We expect BC to remove the class "ExpressCheckoutBlockCollapsed" from
    * the selector when the address fields are replaced. Only when we have
-   * observed thid mutation do we bind an AF widget to the "address_1" field.
+   * observed this mutation do we bind an AF widget to the "address_1" field.
    */
-  var bindWidget = function(addressType, selector, oldValue) {
-    var target = document.querySelector(selector),
-        config = { attributes: true, attributeOldValue: true };
+  var setObserver = function(addressType, elementId, oldValue) {
+    if (window.MutationObserver) {
+
+      /* for modern browsers */
+      var target = document.querySelector("#" + elementId),
+          config = { attributes: true, attributeOldValue: true };
+      var observer = new MutationObserver(function(mutations) {
+        mutations.forEach(function(mutation) {
+          if (mutation.attributeName === "class" && mutation.oldValue.indexOf(oldValue) !== -1) {
+            bindToAddressPanel(addressType, fieldsForAddressType[addressType]["address_1"]);
+            observer.disconnect();
+          }
+        });
+      });
+      observer.observe(target, config);
 
-    var observer = new MutationObserver(function(mutations) {
-      mutations.forEach(function(mutation) {
-        if (mutation.attributeName === "class" && mutation.oldValue.indexOf(oldValue) !== -1) {
+    } else if (window.addEventListener) {
+
+      /* for IE 9 and 10 */
+      var target = document.getElementById(elementId);
+      var listener = function(event) {
+        if (event.attrName.toLowerCase() === "class" && event.prevValue.indexOf(oldValue) !== -1) {
           bindToAddressPanel(addressType, fieldsForAddressType[addressType]["address_1"]);
-          observer.disconnect();
+          target.removeEventListener("DOMAttrModified", listener, false);
         }
-      });
-    });
+      }
+      target.addEventListener("DOMAttrModified", listener, false);
+
+    } else {
+      console.log("AddressFinder Error - please use a more modern browser")
```

Comment:
I think that's a different issue - an expected field is missing, rather
than a lack of an expected API (provided by modern browsers). I'd say keep
it for now.

On Mon, May 9, 2016 at 3:44 PM, fsanggang <notifications@github.com> wrote:

> In addressfinder.js
> <https://github.com/AbleTech/addressfinder-bigcommerce/pull/2#discussion_r62447984>
> :
>
> >          }
> > -      });
> > -    });
> > +      }
> > +      target.addEventListener("DOMAttrModified", listener, false);
> > +
> > +    } else {
> > +      console.log("AddressFinder Error - please use a more modern browser")
>
> OK, and this
> <https://github.com/AbleTech/addressfinder-bigcommerce/blob/develop/addressfinder.js#L202-L206>
> as well?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/AbleTech/addressfinder-bigcommerce/pull/2/files/22d4797ab699a19cb260baff552f367f7a92b7e6#r62447984>
>



-- 
Nigel Ramsay
Principal Consultant
Abletech

04 910 3100
021 323 990
https://abletech.nz
http://nigel.ramsay.org.nz